src/settings/ReadingRoomAccess/ReadingRoomAccess.js

```diff
@@ -74,49 +24,126 @@ const translations = {
 const ReadingRoomAccess = (props) => {
   const intl = useIntl();
   const stripes = useStripes();
+  const { resources } = props;
+
+  // service points defined in the tenant
+  const servicePoints = _.get(resources, ['RRAServicePoints', 'records', 0, 'servicepoints'], []);
+  /**
+   * A reading room can have more than one service points assigned to it.
+   * but a servicepoint cannot be mapped to more than one reading room
+  */
+  const sps = [];
```

Comment:
What about to handle it with base array methods to avoid this tracking of closures? 

Maybe like this:
```
const sps = rrs.reduce((acc, rr) => {
  acc.push(...(rr.servicePoints || []).map((s) => s.value));

  return [...new Set(acc)];
}, []);
```
Or maybe 

```
const sps = rrs
  .map((rr) => rr.servicePoints || [])
  .map((s) => s.value)
  .flat();

const uniqSps = [...new Set(spss)];
```
Note: I didn't check it live with your code, just as an example

Reply:
Quickly check, but it seems not equivalent. I prefer to retain my implementation.